### ISSUE TITLE
False positive of "unused selector" for SCSS and `v-for`

### ISSUE THREAD

**Issue by baseballyama** [future-architect/eslint-plugin-vue-scoped-css#178]
## Summary

False positives seem to occur when using both self-referencing adjacent selector and `v-for`.

## Image

![image](https://user-images.githubusercontent.com/19153718/145703546-03c3fb1b-7c4b-4853-9f30-7b8683fb9cbd.png)

## REPL

```vue
<template>
  <div>
    <div v-for="str in strs" :key="str" class="repl-class">{{ str }}</div>
  </div>
</template>

<script>
import { Component, Vue } from 'vue-property-decorator';

@Component({ components: {} })
export default class REPL extends Vue {
  private strs = ['a', 'b'];
}
</script>

<style lang="scss" scoped>
.repl-class {
  & + .repl-class {
    border-top: #ccc solid 1px;
  }
}
</style>
```

**Comment by ota-meshi** [comment]
Thank you for posting this issue!
That's a bug.

**Comment by baseballyama** [comment]
Thank you for fixing it!
Now It works as I expected!

Thank you so much!

**Comment by ota-meshi** [comment]
Thank you for letting me know.  I'm glad it worked 😆
